### ISSUE TITLE
Cannot get DNS verified for wildcard record

### ISSUE THREAD

**Issue by echodreamz** [rmbolger/Posh-ACME#36]
`PS C:\Users\chris> New-PACertificate '*.REMOVED.com' -AcceptTOS -Contact ssl@REMOVED.com -Verbose
VERBOSE: Using directory https://acme-staging-v02.api.letsencrypt.org/directory
VERBOSE: POST https://acme-staging-v02.api.letsencrypt.org/acme/acct/6159866 with -1-byte payload
VERBOSE: received 349-byte response of content type application/json
VERBOSE: Using account 6159866
VERBOSE: Creating a new order for *.REMOVED.com
VERBOSE: POST https://acme-staging-v02.api.letsencrypt.org/acme/new-order with -1-byte payload
VERBOSE: received 386-byte response of content type application/json
VERBOSE: Publishing DNS challenge for *.REMOVED.com
VERBOSE: Saving TXT record to display when Save-DnsTxtManual is called.
VERBOSE: Saving changes for Manual plugin

Please create the following TXT records:
------------------------------------------
_acme-challenge.REMOVED.com -> abPX3eaRYqLyN9nPIXApJGd7h1x_-3lWCYu73Iskr2k
------------------------------------------

Press any key to continue.:
VERBOSE: Sleeping for 120 seconds while DNS change(s) propagate
VERBOSE: Requesting challenge validations
VERBOSE: POST
https://acme-staging-v02.api.letsencrypt.org/acme/challenge/pv-ga2Vz_4iW9kIjjXPdnW_U68isoBUjvvuhj06ZSHI/130779510
with -1-byte payload
VERBOSE: received 229-byte response of content type application/json
VERBOSE: Saving TXT record to display when Save-DnsTxtManual is called.
VERBOSE: Saving changes for Manual plugin

Please remove the following TXT records:
------------------------------------------
_acme-challenge.REMOVED.com -> abPX3eaRYqLyN9nPIXApJGd7h1x_-3lWCYu73Iskr2k
------------------------------------------

Authorization for *.REMOVED.com returned status 'invalid'.
At C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\2.2.0\Private\Wait-AuthValidation.ps1:33 char:17
+ ...             throw "Authorization for $($auth.fqdn) returned status '$ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Authorization f...atus 'invalid'.:String) [], RuntimeException
    + FullyQualifiedErrorId : Authorization for *.REMOVED.com returned status 'invalid'.`

Doing an nslookup using Google, CloudFlare, OpenDNS and even my local ISP DNS comes back perfect. The TXT record value matches exactly.

`PS C:\Users\chris> nslookup -q=TXT _acme-challenge.REMOVED.com 1.1.1.1
Server:  1dot1dot1dot1.cloudflare-dns.com
Address:  1.1.1.1

Non-authoritative answer:
_acme-challenge.REMOVED.com    text =

        "abPX3eaRYqLyN9nPIXApJGd7h1x_-3lWCYu73Iskr2k"`

I assume this is not a library issue, but something going on on the LE side of things.

**Comment by rmbolger** [comment]
It's hard to say for sure. But yeah, the module appears to be working as intended. This was the ACME server saying that it either couldn't find the record or couldn't find the right value in the record.

How soon after the failure did the nslookup against Cloudflare work? The only thing I can think of is that it hadn't fully propagated after the 2 minute sleep. But it had by the time you ran nslookup.

You could try adding a `-DnsSleep 300` to increase the sleep time to 5 min and see if that fixes anything.

**Comment by echodreamz** [comment]
I checked it about ~30 seconds before doing the command to make sure it was propagated.

**Comment by rmbolger** [comment]
Yeah, that's pretty weird then. What does the error field say when you run the command I posted?

**Comment by echodreamz** [comment]
Weird, the token that the PS script is telling me to create is not the same one returned by the challenge command above.

`type             : dns-01
status           : invalid
error            : @{type=urn:ietf:params:acme:error:caa; detail=CAA record for *.REMOVED.com prevents issuance;
                   status=403}
url              : https://acme-staging-v02.api.letsencrypt.org/acme/challenge/XsoBshWGwBF_x7QgEI4w7AX6XZBrKnL71_au8
                   38n2ik/130804021
token            : wVA8r-Q5Rlr18bNBd4cxXE-LBd6WsaPb2V0DowxmoR4
validationRecord : {@{hostname=REMOVED.com}}`

**Comment by echodreamz** [comment]
I've also fixed the CAA error, but still curious why the tokens are mismatched.

**Comment by rmbolger** [comment]
The token changing is normal if you tried submitting the order again. Once a challenge is invalid, the whole order becomes invalid. So when you run the command again, a new order is created and a new challenge/token with it.
